### ISSUE TITLE
It's too easy to get multi-target target lists in the wrong order

### ISSUE THREAD

**Issue by abadams** [halide/Halide#5050]
E.g. if you say x86-64,x86-64-sse41,x86-64-avx2, then it uses the first thing in the list that matches, which is not what you want, and it uses the last thing in the list for the runtime, which might not run.

**Comment by alexreinking** [comment]
Why does it use the last thing in the list for the runtime? If you generate a standalone runtime with `-r`, it should take `gcd(x86-64,x86-64-sse41,x86-64-avx2) = x86-64`. That behavior should be mirrored for when you aren't doing that.

**Comment by steven-johnson** [comment]
> Why does it use the last thing in the list for the runtime? If you generate a standalone runtime with `-r`, it should take `gcd(x86-64,x86-64-sse41,x86-64-avx2) = x86-64`. 

Yes, we should do that too, but that wouldn't change the fact that the multitarget was likely specified that way by mistake.

**Comment by abadams** [comment]
Oh, then I'm not sure why the above multitarget string produces illegal instructions on machines without avx2.

**Comment by steven-johnson** [comment]
> Oh, then I'm not sure why the above multitarget string produces illegal instructions on machines without avx2.

I interpreted his comment to mean "we *should* be doing that, but aren't currently"

**Comment by abadams** [comment]
There's code in Generator.cpp that attempts to compute a gcd target. It uses Target::get_runtime_compatible_target

**Comment by abadams** [comment]
We should probably add an error when a later target is a strict superset of an earlier target, because that's never useful.
